glommio/src/executor/multitask.rs

```diff
@@ -185,10 +212,71 @@ impl LocalExecutor {
     ///
     /// Returns an option rapping the task.
     pub(crate) fn get_task(&self) -> Option<Runnable> {
-        self.local_queue.pop()
+        let mut run_next = self.run_next.borrow_mut();
+
+        let ret = match (
+            run_next.take(),
+            self.run_next_count.get() < MAX_UNFAIR_SCHEDULING,
+        ) {
+            (Some(runnable), true) => {
+                // run the next task immediately
+                self.run_next_count.replace(self.run_next_count.get() + 1);
+                Some(runnable)
+            }
+            (Some(runnable), false) => {
+                // pop from the queue instead
+                self.run_next_count.replace(0);
+                let ret = self.local_queue.pop();
+                self.local_queue.push(runnable);
```

Comment:
If the queue is empty, then we'll run nothing. Maybe these two lines should be swapped?

Also, maybe we should only reset the counter if the queue is not empty (with a different task)?

Reply:
Uh oh I think you found a bug :sweat_smile:  

The reason I was poping before pushing is that I want to avoid the case where I have to realloc the circular buffer only to pop after just afterwards. I'll make sure I cover the edge-case where I pop nothing!

Reply:
I still want to reset the counter no matter what though. It doesn't matter whether we ended up running the priority runnable or not, the counter should be reset because **we tried** to run something else.